File: recursion_reverse_array.js
reverse_recursion = (array) => {
    n = array.length
    
    if (n <= 1) return array

    return reverse_recursion(array.slice(1)).concat(array[0])
}

reverse_forloop = (array) => {
    to_return = []

    for (let i = 0; i < array.length; i++) {
        to_return.unshift(array[i])
    }

    return to_return
}

time1 = Date.now()

for (let i = 0; i < 998; i++) {
    reverse_recursion(Array(i).fill(0))
}

time2 = Date.now()

for (let i = 0; i < 998; i++) {
    reverse_forloop(Array(i).fill(0))
}

time3 = Date.now()

console.log(`Recursif: ${time2-time1} ms`)
console.log(`For-loop: ${time3-time2} ms`)


console.log( reverse_recursion([0,1,2,3,4,5]) )
console.log( reverse_forloop([0,1,2,3,4,5]) )

File: recursion_reverse_array.py
import time

def reverse_recursion(array: list) -> list:
    n = len(array)
    
    if n <= 1:
        return array.copy()
    
    print(array[n//2:], array[:n//2])
    
    return reverse_recursion(array[n//2:]) + reverse_recursion(array[:n//2])

def reverse_forloop(array: list) -> list:
    to_return: list = []

    for i in array:
        to_return.insert(0, i)

    return to_return


time1 = time.time()

for i in range(998):
    reverse_recursion([0] * i)

time2 = time.time()

for i in range(998):
    reverse_forloop([0] * i)

time3 = time.time()

print(f"Recursif: {time2-time1} s")
print(f"For-loop: {time3-time2} s")

print( reverse_recursion([0,1,2,3,4,5]) )
print( reverse_forloop([0,1,2,3,4,5]) )
